petastorm/benchmark/cli.py

```diff
@@ -0,0 +1,123 @@
+                        help='Size of the shuffling queue used to decorrelate row-group chunks. ')
+
+    parser.add_argument('--min-after-dequeue', type=int, default=None, required=False,
+                        help='Minimum number of elements in a shuffling queue before entries can be read from it. '
+                             'Default value is set to {}%% of the --shuffling-queue-size '
+                             'parameter'.format(100 * DEFAULT_MIN_AFTER_DEQUEUE_TO_QUEUE_SIZE_RATIO))
+
+    parser.add_argument('--experimental-reader', action='store_true', default=False, required=False,
+                        help='If set, a new, experimental version of the reader (ReaderV2) is used.')
+    parser.add_argument('--experimental-decoders-count', type=int, default=3, required=False,
+                        help='Number of decoder threads/processes. Used only if --experimental-reader is set.')
+
+    parser.add_argument('-vv', action='store_true', default=False, help='Sets logging level to DEBUG.')
+    parser.add_argument('-v', action='store_true', default=False, help='Sets logging level to INFO.')
+
+    args = parser.parse_args(args)
+
+    if not args.min_after_dequeue:
+        args.min_after_dequeue = DEFAULT_MIN_AFTER_DEQUEUE_TO_QUEUE_SIZE_RATIO * args.shuffling_queue_size
+
+    return args
+
+
+def _main(args):
+    logging.basicConfig()
+    args = _parse_args(args)
+
+    if args.v:
+        logging.getLogger().setLevel(logging.INFO)
+    if args.vv:
+        logging.getLogger().setLevel(logging.DEBUG)
+
+    if args.experimental_reader:
+        results = reader_v2_throughput(args.dataset_path, args.field_regex, warmup_cycles_count=args.warmup_cycles,
+                                       measure_cycles_count=args.measure_cycles, pool_type=args.pool_type,
+                                       loaders_count=args.workers_count,
+                                       decoders_count=args.experimental_decoders_count, read_method=args.read_method,
+                                       shuffling_queue_size=args.shuffling_queue_size,
+                                       min_after_dequeue=args.min_after_dequeue)
+
+    else:
+        results = reader_throughput(args.dataset_path, args.field_regex, warmup_cycles_count=args.warmup_cycles,
+                                    measure_cycles_count=args.measure_cycles, pool_type=args.pool_type,
+                                    loaders_count=args.workers_count, profile_threads=args.profile_threads,
+                                    read_method=args.read_method, shuffling_queue_size=args.shuffling_queue_size,
+                                    min_after_dequeue=args.min_after_dequeue)
+
+    logger.info('Done')
+    print('Average sample read rate: {:1.2f} samples/sec; RAM {:1.2f} MB (rss); '
+          'CPU {:1.2f}%'.format(results.samples_per_second, results.memory_info.rss / 2**20, results.cpu))
+
+
+def main():
+    _main(sys.argv[1:])
+
+
+if __name__ == '__main__':
```

Comment:
The order of commits seems a bit off since this file was referenced in earlier commits. This isnt a huge deal but could cause trouble if we were to revert based on commit (which we likely wont anyways)